scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +151,147 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
     });
   }
 }
+
+/**
+ * Sorts the release notes dictionary by the patch version
+ */
+export function sortReleaseNotesVersions(markdownByPatchVersion: {
+  [id: string]: string;
+}) {
+  const markdownByPathEntries = Object.entries(markdownByPatchVersion);
+
+  const markdownByPatchSortedArray = markdownByPathEntries.sort(
+    ([version1], [version2]) => {
+      const versionPatch1 = version1.split("rc").slice(0, 1)[0];
+      const versionPatch2 = version2.split("rc").slice(0, 1)[0];
+      const comparison = version2.localeCompare(version1);
+
+      if (versionPatch1 == versionPatch2) {
+        // The release candidates should appear first
+        if (version1.length < version2.length) {
+          return -1;
+        } else if (version1.length > version2.length) {
+          return 1;
+        }
+      }
+      return comparison;
+    },
+  );
+
+  return Object.fromEntries(markdownByPatchSortedArray);
+}
+
+/**
+ * Process the markdown dividing it into small sections for each patch version
+ *
+ * Returns a tuple:
+ *   1. A Set of strings representing all the minor versions we found in the
+ *      markdown.
+ *   2. A dictionary with the corresponding markdown for each patch version
+ *      found.
+ */
+export function extractMarkdownReleaseNotesPatches(
+  markdown: string,
+): [Set<string>, { [id: string]: string }] {
+  const sectionsSplit = markdown.split(/\n## (?=[0-9])/);
+  const sections: string[] = sectionsSplit.slice(1, sectionsSplit.length);
+
+  const minorVersionsFound = new Set<string>();
+  const markdownByPatchVersion: { [id: string]: string } = {};
+
+  for (let section of sections) {
+    const versionPatch = section.split("\n").slice(0, 1)[0];
+    const versionMinor = versionPatch.split(".").slice(0, 2).join(".");
+
+    // Skip legacy release notes
```

Comment:
```suggestion
    // TODO(https://github.com/Qiskit/qiskit/pull/11436). Remove if/when that PR is merged.
    // See that PR for why we have this special logic.
```

This code is a bummer because it breaks the abstraction that releaseNotes.ts had. The code works with any
arbitrary package. But here we're hardcoding in Qiskit-specific logic.

Reply:
You are totally right! I think we should keep the conditional until qiskit `0.45` is not a historical version. Given that we use CI artifacts, we will have the `0.25` version even with the PR merged when regenerating that version. I changed the comment, let me know what you think :+1: